packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts

```diff
@@ -66,9 +66,16 @@ function withCustomResolvers(
       ...config.resolver,
       resolveRequest(...args: Parameters<import('metro-resolver').CustomResolver>) {
         for (const resolver of resolvers) {
-          const resolution = resolver(...args);
-          if (resolution) {
-            return resolution;
+          try {
+            const resolution = resolver(...args);
+            if (resolution) {
+              return resolution;
+            }
+          } catch (err) {
+            // Ignore the `FailedToResolveNameError` error, let the rest resolvers try to handle it
+            if (!(err instanceof Error && err.constructor.name === 'FailedToResolveNameError')) {
+              throw err;
+            }
```

Comment:
This one doesn't seem correct. If you're using a metro resolution then be sure to lock [`metro-resolver` to the same version as `metro`](https://github.com/expo/examples/blob/b7518618feafb6eeaf823687eb4733e93238108e/with-router/package.json#L26-L27).

Reply:
Yeah, I think you are right, so you mean it's better to just use the `instanceof` operator?

Reply:
Oh I misunderstood you before, maybe my comments not clear enough, this piece of code is used to prevent the running `resolver` to throw `FailedToResolveNameError`, because the resolver can not resolve some package is expected, right? 

Sometimes some resolution work is designed to delegate to the following `resolver`, so if the `FailedToResolveNameError` is thrown, then the rest `resolver` will lose the chance to handle it, I think it's not what 

And about why I used this way to detect whether the error is `FailedToResolveNameError`, it's because metro didn't export this error...

Reply:
I'll merge and test myself because I think this is a bit fragile. e.g. `FailedToResolvePathError` is a similar error that should probably be handled the same way.

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63136 created, after that PR has been merged, I will update this PR to use `instanceof` operator

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63136 created, after that PR has been merged, I will update this PR to use `instanceof` operator

Reply:
Can also just pull the code in and add a note to collapse later:
```ts
export type FileAndDirCandidates = {
  dir: FileCandidates;
  file: FileCandidates;
};

/**
 * This is a way to describe what files we tried to look for when resolving
 * a module name as file. This is mainly used for error reporting, so that
 * we can explain why we cannot resolve a module.
 */
export type FileCandidates =
  // We only tried to resolve a specific asset.
  | { type: 'asset'; name: string }
  // We attempted to resolve a name as being a source file (ex. JavaScript,
  // JSON...), in which case there can be several extensions we tried, for
  // example `/js/foo.ios.js`, `/js/foo.js`, etc. for a single prefix '/js/foo'.
  | {
      type: 'sourceFile';
      filePathPrefix: string;
      candidateExts: readonly string[];
    };

type FailedToResolveNameError = Error & {
  dirPaths: string[];
  extraPaths: string[];
};

type FailedToResolvePathError = Error & {
  candidates: FileAndDirCandidates;
};

export function isFailedToResolveNameError(error: any): error is FailedToResolveNameError {
  return !!error && 'extraPaths' in error;
}

export function isFailedToResolvePathError(error: any): error is FailedToResolvePathError {
  return !!error && 'candidates' in error;
}
```